ssm_log_bucket.tf

```
@@ -1,7 +1,7 @@
 locals {
   account_id        = join("", data.aws_caller_identity.current[*].account_id)
   aws_partition     = join("", data.aws_partition.current[*].partition)
-  create_log_bucket = local.enabled && length(var.bucket_id) > 0
+  create_log_bucket = local.enabled && length(var.bucket_id) == 0
   bucket_id         = local.create_log_bucket ? var.bucket_id[0] : module.this.id
```

Tests are failing as this logic was all sorts of wrong it seems. Here is the test failure: 

```
        	            	FatalError{Underlying: error while running command: exit status 1; ╷
        	            	│ Error: Invalid index
        	            	│ 
        	            	│   on ../../ssm_log_bucket.tf line 5, in locals:
        	            	│    5:   bucket_id         = local.create_log_bucket ? var.bucket_id[0] : module.this.id
        	            	│     ├────────────────
        	            	│     │ var.bucket_id is empty list of string
        	            	│ 
        	            	│ The given key does not identify an element in this collection value: the
        	            	│ collection has no elements.
```

This should fix. 

```suggestion
  bucket_id         = local.create_log_bucket ? module.this.id : var.bucket_id[0]
```